base/compiler/ssair/irinterp.jl

```diff
@@ -165,15 +165,15 @@ function reprocess_instruction!(interp::AbstractInterpreter,
     irsv::IRInterpretationState)
     ir = irsv.ir
     if isa(inst, GotoIfNot)
-        cond = argextype(inst.cond, ir)
-        if isa(cond, Const)
+        cond = inst.cond
+        condval = maybe_extract_const_bool(argextype(cond, ir))
+        if condval isa Bool
             function update_phi!(from::Int, to::Int)
                 if length(ir.cfg.blocks[to].preds) == 0
                     return
                 end
                 for idx in ir.cfg.blocks[to].stmts
                     stmt = ir.stmts[idx][:inst]
-                    isa(stmt, Nothing) && continue
```

Comment:
`nothing` is allowed between PhiNodes, so this condition should be retained.

Reply:
Ops, I missed it. Thanks for your pointer.